mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -334,6 +348,11 @@ impl PeerManager {
     }
 
     fn update_prioritized_peers(&self) {
+        // Only do this if it's not a validator
+        if self.role.is_validator() {
+            return;
+        }
```

Comment:
Assuming this is the change that fixes the performance loss to avoid updating the prioritized_peers for validators?  Is this quite costly?

Reply:
So, the truth is there wasn't a significant performance regression directly tied to these pieces.  This portion of code only runs on connect / disconnect, and the reason for ignoring this portion of the code for validators is because priority doesn't matter, all validators are broadcasted to.  It's more costly for validators, because the set of connected validators is uncontrolled (as many validators as exist), vs the set of non-validators which is controlled by the number of outbound connections (3).  I removed this from the previous code, and I've gone ahead and added back in.

Now, the performance regression was only in cluster test, and that was a more nuanced change.  Line 197 in my last change had removed the configurable default_failovers, to what I thought was the same as the default (it was 3).  However, cluster test had set `default_failovers` to 0, to keep the tests from performance regression in the test.  When we decide to change that behavior, we can simply change the config value
```
if priority > self.mempool_config.default_failovers {
    return;
}
```

Reply:
So the broadcast from the full nodes to all nodes was hurting performance in cluster test and now with this limit to the default failovers we get it back?

Reply:
Yes, for the scope of cluster test.